gwdetchar/scattering/core.py

```diff
@@ -107,3 +110,77 @@ def get_fringe_frequency(series, multiplier=2.0):
                         velocity.sample_rate.value)
     fringef.override_unit('Hz')
     return fringef
+
+
+def get_blrms(series, flow=4.0, fhigh=10.0, stride=1, whiten=True,
+              fftlength=4, overlap=2, **kwargs):
+    """Compute the whitened, band-limited RMS of a `TimeSeries`
+
+    Parameters
+    ----------
+    series  : `~gwpy.timeseries.TimeSeries`
+        the input `TimeSeries` data
+
+    flow : `float`, optional
+        lower limit (Hz) of the passband, default: 4.0
+
+    fhigh : `float`, optional
+        upper limit (Hz) of the passband, default: 10.0
+
+    stride: `float`, optional
+        RMS integration length (seconds), default: 1
+
+    whiten : `bool`, optional
+        boolean switch to enable (`True`) or disable (`False`)
+        whitening of the input, default: `True`
+
+    fftlength : `float`, optional
+        FFT integration length (seconds), default: 4
+
+    overlap : `float`, optional
+        FFT overlap length (seconds), default: 2
+
+    **kwargs : `dict`, optional
+        additional keyword arguments to `TimeSeries.whiten`
+
+    Returns
+    -------
+    wblrms : `~gwpy.timeseries.TimeSeries`
+        whitened, band-limited RMS trends of the input `TimeSeries`
+
+    See Also
+    --------
+    ~gwpy.timeseries.TimeSeries.whiten
+        for the underlying whitening scheme
+    ~gwpy.timeseries.TimeSeries.rms
+        for the underlying root-mean-square (RMS) estimation method
+    """
+    if whiten:
+        series = series.whiten(fftlength=fftlength, overlap=overlap, **kwargs)
+    bpseries = series.bandpass(flow, fhigh)
+    return bpseries.rms(stride)
+
+
+def get_segments(series, threshold, name=None):
+    """Generate data-quality segments by thresholding a `TimeSeries`
+
+    Parameters
+    ----------
+    series  : `~gwpy.timeseries.TimeSeries`
+        the input `TimeSeries` data
+
+    threshold : `float`
+        the threshold value for active data-quality segments
+
+    name : `str`, optional
+        name of the data-quality flag, defaults to `series.name`
+
+    Returns
+    --------
+    threshflag : `~gwpy.segments.DataQualityFlag`
+        the populated data-quality flag
+    """
+    thresh = series >= threshold * series.unit
+    name = name or series.name
+    threshflag = thresh.to_dqflag(name)
+    return threshflag
```

Comment:
@siddharth101, digging into the original script, I think we need to gracefully handle the case when no values are above threshold. So, this function should look as follows:

```python
def get_segments(series, threshold, name=None, pad=0):
    """Generate data-quality segments by thresholding a `TimeSeries`

    Parameters
    ----------
    series  : `~gwpy.timeseries.TimeSeries`
        the input `TimeSeries` data

    threshold : `float`
        the threshold value for active data-quality segments

    name : `str`, optional
        name of the data-quality flag, defaults to `series.name`

    Returns
    --------
    threshflag : `~gwpy.segments.DataQualityFlag`
        the populated data-quality flag
    """
    if series.value.max() < threshold:
        return DataQualityFlag(name, known=[series.span])
    else:
        thresh = series >= threshold * series.unit
        name = name or series.name
        threshflag = thresh.to_dqflag(name)
        threshflag.active.protract(pad)
        return threshflag.coalesce()
```

Reply:
@alurban Yup, you are right. I included this for osem segments in the scattering.py script but not for transmon segments. It makes more sense to include it in the function get_segments. I will update it.

Reply:
Thanks @siddharth101, you'll also want to make sure the following top-level import is there:

```python
from gwpy.segments import DataQualityFlag
```